src/app/static/src/app/store/downloadResults/mutations.ts

```diff
@@ -0,0 +1,134 @@
+        state.spectrum.error = action.payload
+        state.spectrum.downloading = false
+
+        if (state.spectrum.statusPollId > -1) {
+            stopPollingSpectrum(state);
+        }
+    },
+
+    [DownloadResultsMutation.SummaryError](state: DownloadResultsState, action: PayloadWithType<Error>) {
+        state.summary.error = action.payload
+        state.summary.downloading = false
+
+        if (state.summary.statusPollId > -1) {
+            stopPollingSummary(state);
+        }
+    },
+
+    [DownloadResultsMutation.CoarseOutputDownloadStatusUpdated](state: DownloadResultsState, action: PayloadWithType<DownloadStatusResponse>) {
+        if (action.payload.done) {
+            stopPollingCoarse(state)
+        }
+        state.coarseOutput.status = action.payload;
+        state.coarseOutput.error = null;
+    },
+
+    [DownloadResultsMutation.SummaryDownloadStatusUpdated](state: DownloadResultsState, action: PayloadWithType<DownloadStatusResponse>) {
+        if (action.payload.done) {
+            stopPollingSummary(state)
+        }
+        state.summary.status = action.payload;
+        state.summary.error = null;
+    },
+
+    [DownloadResultsMutation.SpectrumDownloadStatusUpdated](state: DownloadResultsState, action: PayloadWithType<DownloadStatusResponse>) {
+        if (action.payload.done) {
+            stopPollingSpectrum(state)
+        }
+        state.spectrum.status = action.payload;
+        state.spectrum.error = null;
+    },
+
+    [DownloadResultsMutation.CoarseOutputDownloadComplete](state: DownloadResultsState, action: PayloadWithType<boolean>) {
+        state.coarseOutput.complete = action.payload;
+        state.coarseOutput.downloading = false;
+    },
+
+    [DownloadResultsMutation.SummaryDownloadComplete](state: DownloadResultsState, action: PayloadWithType<boolean>) {
+        state.summary.complete = action.payload;
+        state.summary.downloading = false;
+    },
+
+    [DownloadResultsMutation.SpectrumDownloadComplete](state: DownloadResultsState, action: PayloadWithType<boolean>) {
+        state.spectrum.complete = action.payload;
+        state.spectrum.downloading = false;
+    },
+    [DownloadResultsMutation.PollingStatusStarted](state: DownloadResultsState, action: PayloadWithType<PoolingStarted>) {
+        switch (action.payload.downloadType) {
+            case DOWNLOAD_TYPE.SPECTRUM: {
+                state.spectrum.statusPollId = action.payload.pollId
+                break;
+            }
+            case DOWNLOAD_TYPE.COARSE: {
+                state.coarseOutput.statusPollId = action.payload.pollId
+                break;
+            }
+            case DOWNLOAD_TYPE.SUMMARY: {
+                state.summary.statusPollId = action.payload.pollId
+                break
+            }
+        }
+    }
+};
+
+const stopPollingSpectrum = (state: DownloadResultsState) => {
+    clearInterval(state.spectrum.statusPollId);
```

Comment:
Agree with this. Mutations should really just be modifying the application state.

In fact, I'm not even sure that the action should be doing this (rather than the component) or whether it makes sense to store these IDs in the store at all, but that's more a matter of opinion!

Reply:
We're not very consistent unfortunately - we already have precedent for managing polling in both components and actions e.g. polling for run and calibrate both happen in actions